packages/storybook/src/stories/playground/real-data-demo.stories.tsx

```diff
@@ -34,7 +34,7 @@ const getGridOptions: () => GridOptionsProp = () => {
       action('onSelectionChange', {
         depth: 1,
       })(params),
-    onSortedColumns: (params) => action('onSortedColumns')(params),
+    onColumnsSorted: (params) => action('onColumnsSorted')(params),
```

Comment:
The past it to represent that the action occurs after the sorting has taken place.
The before event is `onColumnsSortingChange`

Reply:
Oh OK, I didn't spot the difference, I thought they were two events for two different states, I didn't realize it was the same state but event raised at different point in time 😇.

What do you think about using two closed names? 

- `onColumnsSortingChange`
- `onColumnsSortingChanged`

Reply:
ATM we have 
- `onColumnsSortingChange`
- `onColumnsSorted`

The 2 names suggested are too similar, it can be confusing I think.

How about we follow the c# convention.
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members

> ✔️ DO give events names with a concept of before and after, using the present and past tenses.
For example, a close event that is raised before a window is closed would be called Closing, and one that is raised after the window is closed would be called Closed.

So in our case, we could do

- `onColumnsSortingChanging` or `onColumnsSortedChanging`, `onColumnsSortChanging`, `onSortChanging` 
- `onColumnsSortingChanged` or `onColumnsSortedChanged`,  `onColumnsSortChanged`, `onSortChanged`

Should we introduce the concept all around? So for pages it would be:
- `onPageChanging` & `onPageChanged`

What if we don't have or see the need for before / after notion in a particular event?
Should we just do it? 

How about click?
`onCellClick` => `onCellClicking`, `onCellClicked` 🤔

Reply:
> c# convention

I think that we can take inspiration from the outside as long as it doesn't conflict with the React convention that have. React should always take **precedence**. React doesn't leave us much choice for the controlled name, it should end with "change".

For the post action, name, do we even need it? Do you have a practical use case. As far as I can push the reflection, I see two cases:

- The data is sorted client-side, there are no delays, the prop has no use cases.
- The data is sorted server-side, the users controls when the data is loaded.

Reply:
- `onColumnsSorting`
- `onColumnsSorted`

No more `change`

> The data is sorted client-side, there are no delays, onColumnsSorted has no use cases.

How do you display a message above the grid after it resorted for example?

Reply:
> How do you display a message above the grid after it resorted for example?

From what I understand, if it's client-side, a developer can display it when onChange is fired. If it's server-side, you control when the data is loaded, a developer can display it when the data is resolved.

Reply:
> If it's server-side, you control when the data is loaded, a developer can display it when the data is resolved.

Yes it is true, but it's a bit less convenient as it would have to tie up the 2 things together. 🤔

Reply:
Yeah, ok sounds fair

Reply:
I was benchmarking on https://react-table-omega.vercel.app/docs/api/useSortBy, it didn't found any conclusion to draw from it.

Reply:
Because it's already fully controllable, they don't abstract any feedback loop.

Reply:
But we can look at the API of v6, the version that made react-table popular: https://github.com/tannerlinsley/react-table/tree/v6

Reply:
I think that as a React user, I would expect a controllable API, something with 3 props: `sortBy`/`defaultSortBy`/`onSortByChange`, something in this order, something backed by [useControlled.js](https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/utils/useControlled.js).

Reply:
I would say the same for the active page state, something like `page/defaultPage/onPageChange`.

Reply:
We could consider this for all the states, checkbox selection, rows selection, cells selection, filtering, etc

Reply:
ok thought about it! Getting rid of the event `onColumnsSorted`

keeping only `onSortingChange` simple and nice what do you think?

Reply:
> keeping only onSortingChange simple and nice what do you think?

I love the direction. The next elements I wonder about are:

- What's the difference between the `sortedColumns` and `sortModel` state? Why do we need `sortedColumns`?
https://github.com/mui-org/material-ui-x/blob/4481b72cf03c6dcd0338d646d1cd6060a4d0b8f4/packages/grid/x-grid-modules/src/hooks/features/useSorting.ts#L120-L125
- Assuming `sortedColumns === sortModel` what about we have `ColumnSortedParams` === `SortModel`?
- What about the [uncontrolled and controlled APIs](https://reactjs.org/docs/glossary.html#controlled-vs-uncontrolled-components)?
- Shouldn't we rename `sortModel` to `defaultSortModal`? to communicate that the state can be changed without changing the prop, it's uncontrolled, and add a `sortModel` prop to control it?
- Shouldn't we rename `onSortingChange(ColumnSortedParams)` to `onSortModelChange(SortModel)` to account for the controlled callback.

Reply:
> Assuming sortedColumns === sortModel what about we have ColumnSortedParams === SortModel?
the idea is to provide all the necessary arguments for the user to do everything it needs without looking to add other props, like in `CellParams` 

> What about the uncontrolled and controlled APIs?
Interesting point. I suggest we start with what we have now and we open an RFC for controlled/uncontrolled. And we can always gather feedbacks first and reassess the need of defaultX prop

> Shouldn't we rename onSortingChange(ColumnSortedParams) to onSortModelChange(SortModel) to account for the controlled callback? 
Sure, happy with that, if we keep the prop `sortModel`

I was thinking of renaming the prop `options.sortModel` with `options.sortBy` what do you think? 
So `onSortingChange` & `sortBy: SortModel` or `onSortModelChange` & `sortModel: SortModel`?

Reply:
taking the later, it's more standard for now.
We can rename later when we clarify a bit more our conventions

Reply:
> I was thinking of renaming the prop options.sortModel with options.sortBy what do you think?

👍 I think that `sortBy` is more descriptive than `sortModel`. We better understand what the option aims to achieve, it sounds better. Should we open an issue to keep track of the idea?

> the idea is to provide all the necessary arguments for the user to do everything it needs without looking to add other props, like in CellParams

No objection here, the reason I have asked is that in order to make the controllable API efficient, for instance https://next--material-ui.netlify.app/components/tree-view/#ControlledTreeView.tsx, you need to be able to wire things:

```tsx
  const [expanded, setExpanded] = React.useState([]);

  const handleToggle = (event, nodeIds) => {
    setExpanded(nodeIds);
  };

  return (
    <TreeView
      expanded={expanded}
      defaultSelected={[]}
      onNodeToggle={handleToggle}
```

P.S. Notice the `onX` and `handleX` naming convention. Notice on how we always expose event. Notice that if you control the component and remove the event feedback loop, nothing happens. What sortModel implements is the uncontrolled behavior with the handle change optional behavior (something an uncontrolled input supports but that we didn't in useControlled.js)

If you want to keep it, I would propose to have the state (sortModel or sortBy once renamed) as the first argument, and the helping values as a second argument, under an object.

> Interesting point. I suggest we start with what we have now and we open an RFC for controlled/uncontrolled. And we can always gather feedbacks first and reassess the need of defaultX prop

Yeah, an RFC sounds like the way to go, we are already off-topic on the pull request!

Reply:
Yes to open an issue with the `sortBy`.